tests/qos/test_buffer.py

```diff
@@ -1919,10 +1958,13 @@ def _get_max_speed_from_list(speed_list_str):
     original_speed = duthost.shell('redis-cli -n 4 hget "PORT|{}" speed'.format(port_to_test))['stdout']
     original_cable_length = duthost.shell('redis-cli -n 4 hget "CABLE_LENGTH|AZURE" {}'.format(port_to_test))['stdout']
     original_pool_size = duthost.shell('redis-cli hget BUFFER_POOL_TABLE:ingress_lossless_pool size')['stdout']
-    if DEFAULT_OVER_SUBSCRIBE_RATIO:
+    if duthost.facts['asic_type'] == 'barefoot':
```

Comment:
Maybe we can introduce a method that returns whether the shared headroom pool is enabled. By doing so the logic can be simplified.
```
def is_shared_headroom_pool_enabled(duthost):
    return DEFAULT_OVER_SUBSCRIBE_RATIO or duthost.facts['asic_type'] in ['barefoot']
```

Reply:
I removed this condition as it is not needed anymore. Thanks!